src/router/index.ts

```diff
@@ -42,14 +48,21 @@ const router = createRouter({
 });
 
 router.beforeEach((to) => {
-  if (
-    to.meta.requiresAuth &&
-    (to.query.userId == undefined || to.query.userId == "")
-  ) {
-    return {
-      name: "403",
-    };
+  /** Before each router, check if the user is a third party and therefore, needs authentication. */
```

Comment:
`requiresAuth` is a navigation guard that is placed on each route. So, here, we have placed `requiresAuth = true` for the quiz route (`/quiz`). So, one cannot access any quiz without the auth. But this allows the flexibility to enable someone to view some other page in the future where we don't need authentication instead of hard-coding the required authentication for every route.